ykrt/src/compile/jitc_yk/opt/mod.rs

```diff
@@ -790,10 +812,13 @@ impl Opt {
                         if off == 0 {
                             self.m.replace(iidx, Inst::Copy(op_iidx));
                         } else {
-                            self.m.replace(
-                                iidx,
-                                Inst::PtrAdd(PtrAddInst::new(Operand::Var(op_iidx), off)),
-                            );
+                            // We can only optimise if the offset can be expressed by an i32.
+                            if let Ok(off) = i32::try_from(off) {
```

Comment:
When we (eventually) replace `i32`, this will slip by the type checker. I'd prefer the "can panic" variant where we just have `Inst::PtrAdd(..., (i32::try_from(off).unwrap())` without the `if let`.

Reply:
Having reviewed this chunk, I don't think this would slip through because constructing the optimised `PtrAddInst` would fail if we had updated the `off` field with a different type: we'd be assigning a `i32` to a different type, so this chunk of code wouldn't compile, right?

We appear not to encounter this case (apart from in the test I devised for it, which I will mark `should_panic` if we really don't want to cover that case for now).